From 0f1f562df3031d56b57199f323891edd7a8f2f51 Mon Sep 17 00:00:00 2001
From: Daniel Dragomir <daniel.dragomir@windriver.com>
Date: Mon, 10 Oct 2022 21:14:45 +0300
Subject: [PATCH] Fix boost depends build with glibc >= 2.34

In glibc >= 2.34 `PTHREAD_STACK_MIN` has been made dynamic which causes
Boost 1.70 to fail to build.

Signed-off-by: Daniel Dragomir <daniel.dragomir@windriver.com>
---
 boost/thread/pthread/thread_data.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/boost/thread/pthread/thread_data.hpp b/boost/thread/pthread/thread_data.hpp
index 588e997f8..f062f843c 100644
--- a/boost/thread/pthread/thread_data.hpp
+++ b/boost/thread/pthread/thread_data.hpp
@@ -57,7 +57,7 @@ namespace boost
 #else
           std::size_t page_size = ::sysconf( _SC_PAGESIZE);
 #endif
-#if PTHREAD_STACK_MIN > 0
+#ifdef PTHREAD_STACK_MIN
           if (size<PTHREAD_STACK_MIN) size=PTHREAD_STACK_MIN;
 #endif
           size = ((size+page_size-1)/page_size)*page_size;
-- 
2.17.1

